src/navigation/Navigator.tsx

```diff
@@ -56,13 +60,25 @@ const styles = (headerHeight?: number) =>
 
 const Stack = createStackNavigator<RoutesParams>()
 
-const Navigator = (): JSX.Element => {
-  const [isPressed, setIsPressed] = React.useState<boolean>(false)
-  const [isHomeButtonPressed, setIsHomeButtonPressed] = React.useState<boolean>(false)
+const Navigator = (): JSX.Element | null => {
+  const [isPressed, setIsPressed] = useState<boolean>(false)
+  const [isHomeButtonPressed, setIsHomeButtonPressed] = useState<boolean>(false)
+  const [waitingForInit, setWaitingForInit] = useState<boolean>(true)
+  const [introShown, setIntroShown] = useState<boolean>(false)
 
   // Set only height for tablets since header doesn't scale auto
   const headerHeight = useTabletHeaderHeight(wp('15%'))
 
+  useEffect(() => {
+    const initialize = async () => {
+      const professions = await AsyncStorage.getSelectedProfessions()
+      setIntroShown(professions !== null)
+    }
+    initialize()
+      .then(() => setWaitingForInit(false))
+      .catch(error => reportError(error))
+  }, [])
```

Comment:
Why do we need a custom useEffect here instead of just using `useReadSelectedProfession`? Then you can just do `const introShown = professions !== null`

Reply:
I see your point here, but not sure if it is possible to do. Profession is null, either when a) intro was not shown before or b) when profession is not yet loaded (when using useReadSelectedProfession). Therefore we need to check for the Promise to return, to set waitingForInti(false) correctly.

Reply:
You can just use `loading` property returned by the hook instead of `waitingForInit` or am I missing something?

Reply:
Nope, you are right, this is working.